apps/passport-server/src/services/e2eeService.ts

```diff
@@ -57,6 +60,62 @@ export class E2EEService {
 
     res.json(undefined satisfies UploadEncryptedStorageResponseValue);
   }
+
+  public async handleChangeBlobKey(
+    request: ChangeBlobKeyRequest,
+    res: Response
+  ): Promise<void> {
+    logger(
+      `[E2EE] Updating ${request.oldBlobKey} to ${request.newBlobKey} for ${request.uuid}`
+    );
+
+    if (
+      !request.newBlobKey ||
+      !request.oldBlobKey ||
+      !request.newSalt ||
+      !request.uuid ||
+      !request.encryptedBlob
+    ) {
+      throw new Error("Missing request fields");
+    }
+
+    // Ensure that old blob key is correct by checking if the row exists
+    const oldRow = await fetchEncryptedStorage(
+      this.context.dbPool,
+      request.oldBlobKey
+    );
+    if (!oldRow) {
+      res
+        .status(401)
+        .json({ error: { name: "PasswordInvalid" }, success: false });
+      return;
+    }
+
+    // Ensure that new salt is different from old salt
+    const commitment = await fetchCommitmentByUuid(
+      this.context.dbPool,
+      request.uuid
+    );
+    if (!commitment) {
+      throw new Error(`User not found with UUID ${request.uuid}`);
+    }
+
+    const { salt: oldSalt } = commitment;
+    if (oldSalt === request.newSalt) {
+      throw new Error("Updated salt must be different than previous salt");
```

Comment:
Same thoughts on status result here.  This feels conceptually similar to the incorrect password check above.